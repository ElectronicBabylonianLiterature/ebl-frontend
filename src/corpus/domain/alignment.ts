export type AlignmentToken =
  | {
      value: string
    }
  | {
      value: string
      alignment: number | null
      variant: ''
    }
  | {
      value: string
      alignment: number | null
      variant: string
      language: string
      isNormalized: boolean
    }

export type Alignment = readonly AlignmentToken[][][]
