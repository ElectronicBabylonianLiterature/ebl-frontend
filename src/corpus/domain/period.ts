export type PeriodModifier = {
  readonly name: string
  readonly displayName: string
}
export const periodModifiers: ReadonlyMap<string, PeriodModifier> = new Map([
  ['None', { name: 'None', displayName: '-' }],
  ['Early', { name: 'Early', displayName: 'Early' }],
  ['Late', { name: 'Late', displayName: 'Late' }],
])

export type Period = {
  readonly name: string
  readonly abbreviation: string
  readonly description: string
  readonly parent?: string
  readonly displayName?: string
}
export const periods: ReadonlyMap<string, Period> = new Map([
  [
    'None',
    {
      name: 'None',
      abbreviation: '',
      description: '',
      displayName: '-',
    },
  ],
  [
    'Ur III',
    {
      name: 'Ur III',
      abbreviation: 'Ur3',
      description: '(ca. 2100–2002 BCE)',
    },
  ],
  [
    'Old Assyrian',
    {
      name: 'Old Assyrian',
      abbreviation: 'OA',
      description: '(ca. 1950–1850 BCE)',
    },
  ],
  [
    'Old Babylonian',
    {
      name: 'Old Babylonian',
      abbreviation: 'OB',
      description: '(ca. 2002–1595 BCE)',
    },
  ],
  [
    'Middle Babylonian',
    {
      name: 'Middle Babylonian',
      abbreviation: 'MB',
      description: '(ca. 1595–1000 BCE)',
    },
  ],
  [
    'Middle Assyrian',
    {
      name: 'Middle Assyrian',
      abbreviation: 'MA',
      description: '(ca. 1595–1000 BCE)',
    },
  ],
  [
    'Hittite',
    {
      name: 'Hittite',
      abbreviation: 'Hit',
      description: '(ca. 1500–1100 BCE)',
    },
  ],
  [
    'Neo-Assyrian',
    {
      name: 'Neo-Assyrian',
      abbreviation: 'NA',
      description: '(ca. 1000–609 BCE)',
    },
  ],
  [
    'Neo-Babylonian',
    {
      name: 'Neo-Babylonian',
      abbreviation: 'NB',
      description: '(ca. 1000–539 BCE)',
    },
  ],
  [
    'Late Babylonian',
    {
      name: 'Late Babylonian',
      abbreviation: 'LB',
      description: '(ca. 539 BCE–ca. 100 CE)',
    },
  ],
  [
    'Persian',
    {
      name: 'Persian',
      abbreviation: 'Per',
      description: '(539–331 BCE)',
      parent: 'Late Babylonian',
    },
  ],
  [
    'Hellenistic',
    {
      name: 'Hellenistic',
      abbreviation: 'Hel',
      description: '(331–141 BCE)',
      parent: 'Late Babylonian',
    },
  ],
  [
    'Parthian',
    {
      name: 'Parthian',
      abbreviation: 'Par',
      description: '(141 BCE–ca. 100 CE)',
      parent: 'Late Babylonian',
    },
  ],
  ['Uncertain', { name: 'Uncertain', abbreviation: 'Unc', description: '' }],
])
