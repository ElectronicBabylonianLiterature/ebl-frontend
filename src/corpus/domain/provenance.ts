export const Provenances = {
  'Standard Text': { name: 'Standard Text', abbreviation: 'Std', parent: null },
  Assyria: { name: 'Assyria', abbreviation: 'Assa', parent: null },
  Aššur: { name: 'Aššur', abbreviation: 'Ašš', parent: 'Assyria' },
  Ḫuzirina: { name: 'Ḫuzirina', abbreviation: 'Huz', parent: 'Assyria' },
  Kalḫu: { name: 'Kalḫu', abbreviation: 'Kal', parent: 'Assyria' },
  Khorsabad: { name: 'Khorsabad', abbreviation: 'Kho', parent: 'Assyria' },
  Nineveh: { name: 'Nineveh', abbreviation: 'Nin', parent: 'Assyria' },
  Tarbiṣu: { name: 'Tarbiṣu', abbreviation: 'Tar', parent: 'Assyria' },
  Babylonia: { name: 'Babylonia', abbreviation: 'Baba', parent: null },
  Babylon: { name: 'Babylon', abbreviation: 'Bab', parent: 'Babylonia' },
  Borsippa: { name: 'Borsippa', abbreviation: 'Bor', parent: 'Babylonia' },
  Cutha: { name: 'Cutha', abbreviation: 'Cut', parent: 'Babylonia' },
  Dilbat: { name: 'Dilbat', abbreviation: 'Dil', parent: 'Babylonia' },
  Isin: { name: 'Isin', abbreviation: 'Isn', parent: 'Babylonia' },
  Kiš: { name: 'Kiš', abbreviation: 'Kiš', parent: 'Babylonia' },
  Larsa: { name: 'Larsa', abbreviation: 'Lar', parent: 'Babylonia' },
  Meturan: { name: 'Meturan', abbreviation: 'Met', parent: 'Babylonia' },
  Nērebtum: { name: 'Nērebtum', abbreviation: 'Nēr', parent: 'Babylonia' },
  Nippur: { name: 'Nippur', abbreviation: 'Nip', parent: 'Babylonia' },
  Sippar: { name: 'Sippar', abbreviation: 'Sip', parent: 'Babylonia' },
  Šaduppûm: { name: 'Šaduppûm', abbreviation: 'Šad', parent: 'Babylonia' },
  Ur: { name: 'Ur', abbreviation: 'Ur', parent: 'Babylonia' },
  Uruk: { name: 'Uruk', abbreviation: 'Urk', parent: 'Babylonia' },
  Periphery: { name: 'Periphery', abbreviation: '', parent: null },
  Alalakh: { name: 'Alalakh', abbreviation: 'Ala', parent: 'Periphery' },
  'Tell el-Amarna': {
    name: 'Tell el-Amarna',
    abbreviation: 'Ama',
    parent: 'Periphery',
  },
  Emar: { name: 'Emar', abbreviation: 'Emr', parent: 'Periphery' },
  Ḫattuša: { name: 'Ḫattuša', abbreviation: 'Hat', parent: 'Periphery' },
  Mari: { name: 'Mari', abbreviation: 'Mar', parent: 'Periphery' },
  Megiddo: { name: 'Megiddo', abbreviation: 'Meg', parent: 'Periphery' },
  Susa: { name: 'Susa', abbreviation: 'Sus', parent: 'Periphery' },
  Ugarit: { name: 'Ugarit', abbreviation: 'Uga', parent: 'Periphery' },
  Uncertain: { name: 'Uncertain', abbreviation: 'Unc', parent: null },
} as const
export type Provenance = typeof Provenances[keyof typeof Provenances]
export const provenances = [
  Provenances['Standard Text'],
  Provenances.Assyria,
  Provenances.Aššur,
  Provenances.Ḫuzirina,
  Provenances.Kalḫu,
  Provenances.Khorsabad,
  Provenances.Nineveh,
  Provenances.Tarbiṣu,
  Provenances.Babylonia,
  Provenances.Babylon,
  Provenances.Borsippa,
  Provenances.Cutha,
  Provenances.Dilbat,
  Provenances.Isin,
  Provenances.Kiš,
  Provenances.Larsa,
  Provenances.Meturan,
  Provenances.Nērebtum,
  Provenances.Nippur,
  Provenances.Sippar,
  Provenances.Šaduppûm,
  Provenances.Ur,
  Provenances.Uruk,
  Provenances.Periphery,
  Provenances.Alalakh,
  Provenances['Tell el-Amarna'],
  Provenances.Emar,
  Provenances.Ḫattuša,
  Provenances.Mari,
  Provenances.Megiddo,
  Provenances.Susa,
  Provenances.Ugarit,
  Provenances.Uncertain,
] as const

export function compareStandardText(
  first: Provenance,
  second: Provenance
): number {
  if (first === second) {
    return 0
  } else if (first === Provenances['Standard Text']) {
    return -1
  } else if (second === Provenances['Standard Text']) {
    return 1
  } else {
    return 0
  }
}

export function compareAssyriaAndBabylonia(
  first: Provenance,
  second: Provenance
): number {
  function isCity(provenance: Provenance): boolean {
    const nonCities: Provenance[] = [
      Provenances['Standard Text'],
      Provenances.Assyria,
      Provenances.Babylonia,
    ]
    return !nonCities.includes(provenance)
  }
  if (isCity(first) && isCity(second)) {
    return 0
  } else if (isCity(first)) {
    return 1
  } else if (isCity(second)) {
    return -1
  } else {
    return compareName(first, second)
  }
}

export function compareName(first: Provenance, second: Provenance): number {
  return first.name.localeCompare(second.name)
}
