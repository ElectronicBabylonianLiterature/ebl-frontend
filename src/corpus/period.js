import type { RecordFactory, RecordOf } from 'immutable'
import { OrderedMap, Record } from 'immutable'

type PeriodModifierProps = { name: string, displayName: string }
const createPeriodModifier: RecordFactory<PeriodModifierProps> = Record({
  name: '',
  displayName: ''
})
export type PeriodModifier = RecordOf<PeriodModifierProps>
export const periodModifiers: OrderedMap<string, PeriodModifier> = OrderedMap({
  None: createPeriodModifier({ name: 'None', displayName: '-' }),
  Early: createPeriodModifier({ name: 'Early', displayName: 'Early' }),
  Late: createPeriodModifier({ name: 'Late', displayName: 'Late' })
})

type PeriodProps = {
  name: string,
  abbreviation: string,
  description: string,
  parent: ?string
}
const createPeriod: RecordFactory<PeriodProps> = Record({
  name: '',
  abbreviation: '',
  description: '',
  parent: null
})
export type Period = RecordOf<PeriodProps>
export const periods: OrderedMap<string, Period> = OrderedMap({
  'Ur III': createPeriod({
    name: 'Ur III',
    abbreviation: 'Ur3',
    description: '(ca. 2100–2002 BCE)'
  }),
  'Old Assyrian': createPeriod({
    name: 'Old Assyrian',
    abbreviation: 'OA',
    description: '(ca. 1950–1850 BCE)'
  }),
  'Old Babylonian': createPeriod({
    name: 'Old Babylonian',
    abbreviation: 'OB',
    description: '(ca. 2002–1595 BCE)'
  }),
  'Middle Babylonian': createPeriod({
    name: 'Middle Babylonian',
    abbreviation: 'MB',
    description: '(ca. 1595–1000 BCE)'
  }),
  'Middle Assyrian': createPeriod({
    name: 'Middle Assyrian',
    abbreviation: 'MA',
    description: '(ca. 1595–1000 BCE)'
  }),
  Hittite: createPeriod({
    name: 'Hittite',
    abbreviation: 'Hit',
    description: '(ca. 1500–1100 BCE)'
  }),
  'Neo-Assyrian': createPeriod({
    name: 'Neo-Assyrian',
    abbreviation: 'NA',
    description: '(ca. 1000–609 BCE)'
  }),
  'Neo-Babylonian': createPeriod({
    name: 'Neo-Babylonian',
    abbreviation: 'NB',
    description: '(ca. 1000–539 BCE)'
  }),
  'Late Babylonian': createPeriod({
    name: 'Late Babylonian',
    abbreviation: 'LB',
    description: '(ca. 539 BCE–ca. 100 CE)'
  }),
  Persian: createPeriod({
    name: 'Persian',
    abbreviation: 'Per',
    description: '(539–331 BCE)',
    parent: 'Late Babylonian'
  }),
  Hellenistic: createPeriod({
    name: 'Hellenistic',
    abbreviation: 'Hel',
    description: '(331–141 BCE)',
    parent: 'Late Babylonian'
  }),
  Parthian: createPeriod({
    name: 'Parthian',
    abbreviation: 'Par',
    description: '(141 BCE–ca. 100 CE)',
    parent: 'Late Babylonian'
  }),
  Uncertain: createPeriod({
    name: 'Uncertain',
    abbreviation: 'Unc',
    description: ''
  })
})
