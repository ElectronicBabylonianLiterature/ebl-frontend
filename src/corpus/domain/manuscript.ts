import Reference from 'bibliography/domain/Reference'
import produce, { Draft, immerable } from 'immer'
import _ from 'lodash'
import { Period, PeriodModifier, periodModifiers, periods } from './period'
import { Provenance, provenances } from './provenance'

export interface ManuscriptType {
  readonly name: string
  readonly abbreviation: string
  readonly displayName?: string
}

export const types: ReadonlyMap<string, ManuscriptType> = new Map([
  ['None', { name: 'None', abbreviation: '', displayName: '-' }],
  ['Library', { name: 'Library', abbreviation: '' }],
  ['School', { name: 'School', abbreviation: 'Sch' }],
  ['Varia', { name: 'Varia', abbreviation: 'Var' }],
  ['Commentary', { name: 'Commentary', abbreviation: 'Com' }],
  ['Quotation', { name: 'Quotation', abbreviation: 'Quo' }],
  ['Excerpt', { name: 'Excerpt', abbreviation: 'Ex' }],
  ['Parallel', { name: 'Parallel', abbreviation: 'Par' }],
])

export class Manuscript {
  readonly [immerable] = true
  readonly id: number | undefined | null = null
  readonly siglumDisambiguator: string = ''
  readonly museumNumber: string = ''
  readonly accession: string = ''
  readonly periodModifier: PeriodModifier = periodModifiers.get('None') || {
    name: 'None',
    displayName: '-',
  }
  readonly period: Period = periods.get('Neo-Assyrian') || {
    name: 'Neo-Assyrian',
    abbreviation: 'NA',
    description: '(ca. 1000â€“609 BCE)',
  }
  readonly provenance: Provenance = provenances.get('Nineveh') || {
    name: 'Nineveh',
    abbreviation: 'Nin',
    parent: 'Assyria',
  }
  readonly type: ManuscriptType = types.get('Library') || {
    name: 'Library',
    abbreviation: '',
  }
  readonly notes: string = ''
  readonly colophon: string = ''
  readonly unplacedLines: string = ''
  readonly references: readonly Reference[] = []

  get siglum(): string {
    return [
      _.get(this, 'provenance.abbreviation', ''),
      _.get(this, 'period.abbreviation', ''),
      _.get(this, 'type.abbreviation', ''),
      this.siglumDisambiguator,
    ].join('')
  }
}

export function createManuscript(data: Partial<Manuscript>): Manuscript {
  return produce(new Manuscript(), (draft: Draft<Manuscript>) => {
    _.assign(draft, data)
  })
}
