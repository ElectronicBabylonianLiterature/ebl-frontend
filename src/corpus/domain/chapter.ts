import _ from 'lodash'
import { immerable } from 'immer'
import flow from 'lodash/fp/flow'
import flatMap from 'lodash/fp/flatMap'
import map from 'lodash/fp/map'
import Cite from 'citation-js'
import removeMd from 'remove-markdown'
import { Parser } from 'html-to-react'
import { LineNumber, OldLineNumber } from 'transliteration/domain/line-number'
import { MarkupPart } from 'transliteration/domain/markup'
import { ChapterAlignment } from './alignment'
import { Line, ManuscriptLine } from './line'
import { Manuscript } from './manuscript'
import { TextId, textIdToDoiString } from 'transliteration/domain/text-id'
import TranslationLine from 'transliteration/domain/translation-line'
import { ChapterId, defaultName } from 'transliteration/domain/chapter-id'
import { NoteLine } from 'transliteration/domain/note-line'
import { ParallelLine } from 'transliteration/domain/parallel-line'
import { Token } from 'transliteration/domain/token'
import { ManuscriptLineDisplay } from 'corpus/domain/line-details'
import { stageToAbbreviation } from 'corpus/domain/period'

export class Chapter {
  readonly [immerable] = true

  constructor(
    readonly textId: TextId,
    readonly textHasDoi: boolean,
    readonly classification: string,
    readonly stage: string,
    readonly version: string,
    readonly name: string,
    readonly order: number,
    readonly manuscripts: ReadonlyArray<Manuscript>,
    readonly uncertainFragments: ReadonlyArray<string>,
    readonly lines: ReadonlyArray<Line>
  ) {}

  get id(): ChapterId {
    return {
      textId: this.textId,
      stage: this.stage,
      name: this.name,
    }
  }

  get alignment(): ChapterAlignment {
    return new ChapterAlignment(
      this.lines.map((line) =>
        line.variants.map((variant) => variant.alignment)
      )
    )
  }

  getSiglum(manuscriptLine: ManuscriptLine): string {
    const manuscript = this.manuscripts.find(
      (candidate) => candidate.id === manuscriptLine.manuscriptId
    )
    if (manuscript) {
      return manuscript.siglum
    } else {
      return `<unknown ID: ${manuscriptLine.manuscriptId}>`
    }
  }
}

export interface LineVariantDisplay {
  readonly reconstruction: ReadonlyArray<Token>
  readonly note: NoteLine | null
  readonly manuscripts: ReadonlyArray<ManuscriptLineDisplay>
  readonly parallelLines: ReadonlyArray<ParallelLine>
  readonly intertext: ReadonlyArray<MarkupPart>
}

export interface LineDisplay {
  readonly number: LineNumber
  readonly oldLineNumbers: ReadonlyArray<OldLineNumber>
  readonly isSecondLineOfParallelism: boolean
  readonly isBeginningOfSection: boolean
  readonly translation: ReadonlyArray<TranslationLine>
  readonly variants: ReadonlyArray<LineVariantDisplay>
}

export interface Author {
  readonly name: string
  readonly prefix: string
  readonly role: 'EDITOR' | 'REVISION'
  readonly orcidNumber: string
}

export interface Translator {
  readonly name: string
  readonly prefix: string
  readonly orcidNumber: string
  readonly language: string
}

export interface Record {
  readonly authors: ReadonlyArray<Author>
  readonly translators: ReadonlyArray<Translator>
  readonly publicationDate: string
}

export class ChapterDisplay {
  readonly [immerable] = true

  constructor(
    readonly id: ChapterId,
    readonly textHasDoi: boolean,
    readonly textName: string,
    readonly isSingleStage: boolean,
    readonly title: ReadonlyArray<MarkupPart>,
    readonly lines: ReadonlyArray<LineDisplay>,
    readonly record: Record,
    readonly atf: string
  ) {}

  get languages(): Set<string> {
    return flow(
      flatMap<LineDisplay, TranslationLine>((line) => line.translation),
      map((translation) => translation.language),
      (languages) => new Set(languages)
    )(this.lines)
  }

  get isPublished(): boolean {
    return (
      !_.isEmpty(this.record.publicationDate) && !_.isEmpty(this.record.authors)
    )
  }

  get uniqueIdentifier(): string {
    return this.idParts.join(' ')
  }

  get fullName(): string {
    const showName = this.id.name !== defaultName || this.isSingleStage
    return [
      removeMd(this.textName),
      'Chapter',
      !this.isSingleStage ? this.id.stage : '',
      showName ? this.id.name : '',
    ]
      .filter(_.negate(_.isEmpty))
      .join(' ')
  }

  get url(): string {
    const idPartsAbbreviation = [
      ...this.idParts.slice(0, 3),
      stageToAbbreviation(this.idParts[3]),
      this.idParts[4],
    ]
    return `https://www.ebl.lmu.de/corpus/${idPartsAbbreviation
      .map(encodeURIComponent)
      .join('/')}`
  }

  get doi(): string {
    return this.textHasDoi ? textIdToDoiString(this.id.textId) : ''
  }

  get parsedCitation(): JSX.Element {
    return new Parser().parse(
      this.citation.format('bibliography', {
        format: 'html',
        template: 'citation-apa',
      })
    )
  }

  get citation(): Cite {
    const issued = new Date(this.record.publicationDate)
    const now = new Date()
    return new Cite({
      id: this.uniqueIdentifier,
      type: 'article-journal',
      author: this.record.authors.map((author) => ({
        family: author.name,
        given: author.prefix,
      })),
      accessed: {
        'date-parts': [[now.getFullYear(), now.getMonth() + 1, now.getDate()]],
      },
      issued: {
        'date-parts': [
          [issued.getFullYear(), issued.getMonth() + 1, issued.getDate()],
        ],
      },
      title: this.fullName,
      'container-title': 'electronic Babylonian Literature',
      URL: this.url,
      ...(this.doi && { DOI: this.doi }),
    })
  }

  private get idParts(): readonly [string, number, number, string, string] {
    return [
      this.id.textId.genre,
      this.id.textId.category,
      this.id.textId.index,
      this.id.stage,
      this.id.name,
    ]
  }

  getTranslatorsOf(language: string): Translator[] {
    return this.record.translators.filter(
      (translator) => translator.language === language
    )
  }
}
