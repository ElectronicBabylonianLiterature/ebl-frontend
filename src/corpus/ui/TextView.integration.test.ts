import Chance from 'chance'
import { ChapterId } from 'corpus/domain/chapter'
import { ManuscriptTypes } from 'corpus/domain/manuscript'
import { PeriodModifiers, Periods } from 'corpus/domain/period'
import { Provenances } from 'corpus/domain/provenance'
import AppDriver from 'test-support/AppDriver'
import { referenceDtoFactory } from 'test-support/bibliography-fixtures'
import FakeApi from 'test-support/FakeApi'
import { joinDtoFactory } from 'test-support/join-fixtures'
import { textDto } from 'test-support/test-corpus-text'

const chance = new Chance('text view integration test')
const manuscriptsDto = [
  {
    id: 1,
    siglumDisambiguator: '1',
    museumNumber: 'BM.X',
    accession: '',
    periodModifier: PeriodModifiers.None.name,
    period: Periods['Old Babylonian'].name,
    provenance: Provenances.Assyria.name,
    type: ManuscriptTypes.School.name,
    notes: '',
    colophon: '1. kur',
    unplacedLines: '1. bu',
    references: referenceDtoFactory.buildList(
      1,
      {},
      { transient: { chance: chance } }
    ),
    joins: [
      [
        joinDtoFactory.build(
          {
            isChecked: false,
            isInFragmentarium: true,
          },
          { transient: { chance: chance } }
        ),
        joinDtoFactory.build(
          {
            isChecked: false,
            isInFragmentarium: false,
          },
          { transient: { chance: chance } }
        ),
      ],
      [
        joinDtoFactory.build(
          {
            isChecked: true,
            isInFragmentarium: true,
          },
          { transient: { chance: chance } }
        ),
        joinDtoFactory.build(
          {
            isChecked: true,
            isInFragmentarium: false,
          },
          { transient: { chance: chance } }
        ),
      ],
    ],
    isInFragmentarium: false,
  },
]

let fakeApi: FakeApi
let appDriver: AppDriver

afterEach(() => {
  fakeApi.verifyExpectations()
})

beforeEach(() => {
  fakeApi = new FakeApi()
  fakeApi.expectText(textDto)
  appDriver = new AppDriver(fakeApi.client).withPath(
    `/corpus/${textDto.genre}/${textDto.category}/${textDto.index}`
  )
})

test('With session', async () => {
  await appDriver.withSession().render()
  await appDriver.waitForText('Introduction')
  expect(appDriver.getView().container).toMatchSnapshot()
})

describe('Chapter', () => {
  beforeEach(async () => {
    await appDriver.withSession().render()
    await appDriver.waitForText(/Chapters/)
    appDriver.click(/Chapters/)
    await appDriver.waitForText(textDto.chapters[0].name)
  })

  test('Show chapter', async () => {
    expect(appDriver.getView().container).toMatchSnapshot()
  })

  test('Show list of manuscripts', async () => {
    const chapterId: ChapterId = {
      textId: {
        genre: textDto.genre,
        category: textDto.category,
        index: textDto.index,
      },
      stage: textDto.chapters[0].stage,
      name: textDto.chapters[0].name,
    }
    fakeApi
      .expectManuscripts(chapterId, manuscriptsDto)
      .expectExtantLines(chapterId, {
        AssaOBSch1: {
          'o iii': [
            {
              lineNumber: {
                number: 1,
                hasPrime: false,
                prefixModifier: null,
                suffixModifier: null,
              },
              isSideBoundary: true,
            },
            {
              lineNumber: {
                number: 2,
                hasPrime: false,
                prefixModifier: null,
                suffixModifier: null,
              },
              isSideBoundary: false,
            },
          ],
        },
      })
    appDriver.click(/List of Manuscripts/)
    await appDriver.waitForText(/^o iii/)
    expect(appDriver.getView().container).toMatchSnapshot()
  })
})

test('Without session', async () => {
  await appDriver.render()
  await appDriver.waitForText('Please log in to view the text.')
  expect(appDriver.getView().container).toMatchSnapshot()
})
