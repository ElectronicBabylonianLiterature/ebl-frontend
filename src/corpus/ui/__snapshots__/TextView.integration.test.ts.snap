// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With session 1`] = `
<div>
  <header
    class="Header"
  >
    <nav
      class="navbar navbar-expand-md navbar-light"
    >
      <div
        class="container"
      >
        <span
          class="navbar-brand"
        >
          <a
            href="https://www.lmu.de"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              class="Header__corporate-logo img-fluid"
              src="LMU_Logo.svg"
            />
          </a>
        </span>
        <a
          class="active navbar-brand"
          href="/"
          title="electronic Babylonian Literature (eBL)"
        >
          <h1
            class="Header__title"
          >
            <span
              class="Header__title-main"
            >
              electronic
              <br />
              Babylonian
              <br />
              Literature
            </span>
            <small
              class="Header__title-abbreviation"
            >
              eBL
            </small>
          </h1>
        </a>
        <div
          class="navbar-collapse collapse"
          id="Header-1"
        >
          <div
            class="mx-auto navbar-nav"
          >
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/signs"
                href="/signs"
              >
                Signs
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/dictionary"
                href="/dictionary"
              >
                Dictionary
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="active nav-link"
                data-rb-event-key="/corpus"
                href="/corpus"
              >
                Corpus
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/fragmentarium"
                href="/fragmentarium"
              >
                Fragmentarium
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/bibliography"
                href="/bibliography"
              >
                Bibliography
              </a>
            </div>
          </div>
          <span
            class="navbar-text"
          >
            <button
              class="btn btn-outline-secondary btn-sm"
              type="button"
            >
              Logout 
              Test
            </button>
          </span>
        </div>
        <button
          aria-controls="Header-1"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
      </div>
    </nav>
  </header>
  <section
    class="App-content"
  >
    <header
      class="App-content__header"
    >
      <nav
        aria-label="breadcrumb"
      >
        <ol
          class="breadcrumb"
        >
          <li
            class="breadcrumb-item active"
          >
            <a
              href="/"
            >
              eBL
            </a>
          </li>
          <li
            class="breadcrumb-item active"
          >
            <a
              href="/corpus"
            >
              Corpus
            </a>
          </li>
          <li
            aria-current="page"
            class="breadcrumb-item active"
          >
            I.1 Palm and Vine
          </li>
        </ol>
      </nav>
      <div
        class="float-right btn-toolbar"
        role="toolbar"
      />
      <h2>
        I.1 Palm and Vine
      </h2>
    </header>
    <section
      class="text-view__section"
    >
      <h3
        class="text-view__section-heading"
      >
        Introduction
      </h3>
      <div
        class="text-view__markdown"
      >
        <p>
          This is a 
          <em>
            test text
          </em>
          .
        </p>
      </div>
    </section>
    <section
      class="text-view__section"
    >
      <h3
        class="text-view__section-heading"
      >
        Chapters
      </h3>
      <div
        class="nav nav-tabs"
      >
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rb-event-key="/corpus/L/1/1/Old Babylonian/The Only Chapter"
            href="/corpus/L/1/1/Old Babylonian/The Only Chapter"
          >
            Old Babylonian
             
            The Only Chapter
          </a>
        </div>
      </div>
    </section>
    <section
      class="text-view__section"
    >
      <h3
        aria-controls="collapse-2"
        class="text-view__section-heading text-view__section-heading--collapse"
      >
        Colophons
         
        <i
          aria-expanded="false"
          class="text-view__collapse-indicator fas fa-caret-right"
        />
      </h3>
    </section>
    <section
      class="text-view__section"
    >
      <h3
        aria-controls="collapse-3"
        class="text-view__section-heading text-view__section-heading--collapse"
      >
        Unplaced Lines
         
        <i
          aria-expanded="false"
          class="text-view__collapse-indicator fas fa-caret-right"
        />
      </h3>
    </section>
  </section>
</div>
`;

exports[`Without session 1`] = `
<div>
  <header
    class="Header"
  >
    <nav
      class="navbar navbar-expand-md navbar-light"
    >
      <div
        class="container"
      >
        <span
          class="navbar-brand"
        >
          <a
            href="https://www.lmu.de"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              class="Header__corporate-logo img-fluid"
              src="LMU_Logo.svg"
            />
          </a>
        </span>
        <a
          class="active navbar-brand"
          href="/"
          title="electronic Babylonian Literature (eBL)"
        >
          <h1
            class="Header__title"
          >
            <span
              class="Header__title-main"
            >
              electronic
              <br />
              Babylonian
              <br />
              Literature
            </span>
            <small
              class="Header__title-abbreviation"
            >
              eBL
            </small>
          </h1>
        </a>
        <div
          class="navbar-collapse collapse"
          id="Header-4"
        >
          <div
            class="mx-auto navbar-nav"
          >
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/signs"
                href="/signs"
              >
                Signs
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/dictionary"
                href="/dictionary"
              >
                Dictionary
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="active nav-link"
                data-rb-event-key="/corpus"
                href="/corpus"
              >
                Corpus
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/fragmentarium"
                href="/fragmentarium"
              >
                Fragmentarium
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/bibliography"
                href="/bibliography"
              >
                Bibliography
              </a>
            </div>
          </div>
          <span
            class="navbar-text"
          >
            <button
              class="btn btn-outline-secondary btn-sm"
              type="button"
            >
              Login
            </button>
          </span>
        </div>
        <button
          aria-controls="Header-4"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
      </div>
    </nav>
  </header>
  <section
    class="App-content"
  >
    <header
      class="App-content__header"
    >
      <nav
        aria-label="breadcrumb"
      >
        <ol
          class="breadcrumb"
        >
          <li
            class="breadcrumb-item active"
          >
            <a
              href="/"
            >
              eBL
            </a>
          </li>
          <li
            class="breadcrumb-item active"
          >
            <a
              href="/corpus"
            >
              Corpus
            </a>
          </li>
          <li
            aria-current="page"
            class="breadcrumb-item active"
          >
            I.1 Palm and Vine
          </li>
        </ol>
      </nav>
      <div
        class="float-right btn-toolbar"
        role="toolbar"
      />
      <h2>
        I.1 Palm and Vine
      </h2>
    </header>
    <p>
      Please log in to view the text.
    </p>
  </section>
</div>
`;
